src/keygen/participant.rs

```diff
@@ -594,9 +592,7 @@ mod tests {
 
     fn is_keygen_done(quorum: &[KeygenParticipant], keygen_identifier: Identifier) -> Result<()> {
         for participant in quorum {
-            if participant.is_keygen_done(keygen_identifier).is_err() {
-                return bail!("Keygen not done");
-            }
+            participant.is_keygen_done(keygen_identifier)?;
```

Comment:
Function names like `is_*` or `has_*` imply they return a bool. Since we are no longer returning a bool we should rename the function to something like `check_keygen_done`. Same for other functions you changed.